Lib/glyphsLib/classes.py

```diff
@@ -1418,6 +1418,36 @@ def connection(self):
     def selected(self):
         raise OnlyInGlyphsAppError
 
+    def _encode_dict_as_string(self, value):
+        """Takes the PLIST string of a dict, and returns the same string
+        encoded such that it can be included in the string representation
+        of a GSNode."""
+        # Strip the first and last newlines
+        if value.startswith('{\n'):
+            value = '{' + value[2:]
+        if value.endswith('\n}'):
+            value = value[:-2] + '}'
+        value = value.replace('"', '\\"')
+        value = value.replace('\n', '\\n')
+        return value
```

Comment:
I wonder if instead of all this string copies and consecutive calls to `.replace()` would be more efficient to use the re.sub approach (with a callable `repl` function that takes the match object and returns a string, like the `_decode_dict_as_string` does)

Reply:
It looks like re.sub is slower (maybe because I tested with a small input string, but I guess that most strings in that case will have only a name for the node)

https://gist.github.com/belluzj/9f7281384190a9988e17d98d5e3d1686

Reply:
It’s good that you timed it!
Does it make any difference if instead of calling re.sub you first re.compile and then call sub method on the compiled regex object?

Reply:
ok good!